pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -60,16 +64,16 @@ var _ controller.Reconciler = (*Reconciler)(nil)
 
 // Reconcile compares the actual state with the desired, and attempts to
 // converge the two.
-func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
+func (c *Reconciler) Reconcile(ctx context.Context, key string) (returnError error) {
 	c.Logger.Infof("event-listener-reconcile %s", key)
 	// Convert the namespace/name string into a distinct namespace and name
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		c.Logger.Errorf("invalid resource key: %s", key)
-		return nil
+		return err
```

Comment:
I think we should go back to returning nil (or an instance of PermanentError) here. Otherwise, we will continue to try to reconcile with the invalid resource key:https://github.com/tektoncd/triggers/blob/master/vendor/knative.dev/pkg/controller/controller.go#L330

Reply:
So return nil, which will not be considered a reconcile failure?